src/zip.ts

```diff
@@ -0,0 +1,42 @@
+import { purry } from './purry'
+
+/**
+ * Creates a new list from two supplied lists by pairing up equally-positioned items.
+ * The length of the returned list will match the shortest of the two inputs.
+ * @param first the first input list
+ * @param second the second input list
+ * @signature
+ *   R.zip(first, second)
+ * @example
+ *   R.zip([1, 2], ['a', 'b']) // => [1, 'a'], [2, 'b']
+ * @data_first
+ * @category Array
+ */
+export function zip<F extends unknown, S extends unknown>(first: Array<F>, second: Array<S>): Array<[F, S]>
```

Comment:
ah, we now need to make that `ReadonlyArray` please. Otherwise, we cannot input readonly array, e.g. derived from `const assertions`, see this [playground](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAMwK4DsDGMCWEVwF6ZgA8AYnKDMCgCYDOcqA1ihAO4oA0cAyuSJTXpMW7AHwAKBJii0YALjgBBKFACGAT1KiutYGhzUFytZu6iAlEZUaiAbRJduAXVFwA3gFgAUHF9wA9P5wskgARnCYALZgADZcrMBwODHqcGiqUImqoRBI8DAAFokw6mDAtN5+cJkwSFC4tk5wqvSqKOreAL7e3voosogQEHAAvHC2AOQIQxNcE6EZE00taTiyvWvwC3ij4wCMXABMy-R9614b-fCZtEgx8GMEYJJDXNvmQA)

The good thing about `readonly` is that you can always pass a mutable Array into it, but it doesn't work the other way around. The return value should still be an `Array`.

Maybe we could add some type-level tests like we [do them here](https://github.com/remeda/remeda/blob/17e9b0e2985d26bc6689a77a4e192baa80fcc29f/src/reverse.test.ts#L11-L15) ?

Reply:
Added those.  Thank you for your patience and the detailed reviews.  I have a few PRs open here--I'm planning to apply the takeaways here to those PRs since I think many of these items will apply.